### DISCUSSION TITLE
Add module level entrypoint

### DISCUSSION THREAD

**Discussion by pohmelie** [encode/httpx#3216] [Ideas]
I was surprised when discover that httpx can't be invoked as cli via `python -m httpx`, only as an registered entrypoint as `httpx`. The first one is everywhere in python world: ruff, mypy, black, pip, pytest, etc.

**Comment by lovelydinosaur** [comment]
Okay, yeah this could be valid.

Looking at [`flask/__main__.py`](https://github.com/pallets/flask/blob/main/src/flask/__main__.py) and [`httpx/__init__.py`](https://github.com/encode/httpx/blob/master/httpx/__init__.py#L14-L26) I think we'd need this?...

**__main__.py**...

```python
try:
    from ._main import main
except ImportError:  # pragma: no cover

    def main() -> None:  # type: ignore
        import sys

        print(
            "The httpx command line client could not run because the required "
            "dependencies were not installed.\nMake sure you've installed "
            "everything with: pip install 'httpx[cli]'"
        )
        sys.exit(1)

main()
```

**Reply by pohmelie** [reply]
It looks like «yes»
